nservicebus/messaging/databus.md

```diff
@@ -66,9 +66,46 @@ snippet:MessageWithLargePayloadUsingConvention
 
 ## DataBus attachments cleanup
 
-NServiceBus DataBus implementations currently behave differently with regard to cleanup of physical attachments used to transfer data properties. `FileShareDataBus` **does not** remove physical attachments once the message is gone. `AzureDataBus` **does** remove Azure storage blobs used for physical attachments once the message is gone.
+NServiceBus DataBus implementations currently behaves differently with regard to cleanup of physical attachments used to transfer data properties depending on the implementation used.
 
 
+### Reasons why attachments are not removed by default
+
+Automatically removing these attachments can cause problems in many situations. For example:
+
+* The supported data bus implementations do not participate in distributed transactions. If for some reason, the message handler throws an exception and the transaction rolls back, the delete operation on the attachment cannot be rolled back. Therefore, when the message is retried, the attachment will no longer be present causing additional problems.
```

Comment:
I would mention the error q/ retrying from SP since that can happen "days" after the message failed